src/core/StelSkyCultureMgr.cpp

```diff
@@ -756,3 +770,224 @@ QString StelSkyCultureMgr::skyCultureI18ToDirectory(const QString& cultureName)
+	emit infoLabelStyleChanged(style);
+}
+
+void StelSkyCultureMgr::setInfoLabelStyle(const QString &style)
+{
+	setInfoLabelStyle(convertCulturalDisplayStyleFromCSVstring(style));
+}
+
+QString StelSkyCultureMgr::createCulturalLabel(const StelObject::CulturalName &cName,
+					       const StelObject::CulturalDisplayStyle style,
+					       const QString &commonNameI18n,
+					       const QString &abbrevI18n) const
+{
+	// At least while many fields have not been filled, we should create a few fallbacks
+	// If native contains non-Latin glyphs, pronounce or transliteration is mandatory.
+	QString pronounceStr=(cName.pronounceI18n.isEmpty() ? cName.pronounce : cName.pronounceI18n);
+	QString nativeOrPronounce = (cName.native.isEmpty() ? cName.pronounceI18n : cName.native);
+	QString pronounceOrNative = (cName.pronounceI18n.isEmpty() ? cName.native : cName.pronounceI18n);
+	QString translitOrPronounce = (cName.transliteration.isEmpty() ? pronounceStr : cName.transliteration);
+
+	// If you call this with an actual argument abbrevI18n, you really only want a short label.
+	if (flagUseAbbreviatedNames && !abbrevI18n.isNull())
+		return (abbrevI18n.startsWith('.') ? QString() : abbrevI18n);
+
+	const int styleInt=int(style);
+	QString label;
+	switch (style)
+	{
+		case StelObject::CulturalDisplayStyle::Native: // native if available. fallback to pronounce and english entries
+			return cName.native.isEmpty() ? (cName.pronounceI18n.isEmpty() ? cName.translatedI18n : cName.pronounceI18n) : cName.native;
+		case StelObject::CulturalDisplayStyle::Pronounce: // pronounce if available. fallback to native
+			return pronounceOrNative;
+		case StelObject::CulturalDisplayStyle::Translit:
+			return translitOrPronounce;
+		case StelObject::CulturalDisplayStyle::Translated:
+			return (cName.translatedI18n.isEmpty() ? (pronounceStr.isEmpty() ? cName.native : pronounceStr) : cName.translatedI18n);
+		case StelObject::CulturalDisplayStyle::IPA: // really only IPA?
+			return cName.IPA;
+		case StelObject::CulturalDisplayStyle::NONE: // fully non-cultural!
+		case StelObject::CulturalDisplayStyle::Modern:
+			return commonNameI18n;
+		default:
+			break;
+	}
+	// simple cases done. Now build-up label of form "primary [common transliteration aka pronounce, scientific transliteration, IPA] (translation) <modern>"
+	// "primary" is usually either native or one of the reading aids, or translation or even modern when other options are switched off.
+	// Rules:
+	// Styles with Native_* start with just native, but we must fallback to other strings if native is empty.
+	// Styles with Pronounce_* start with Pronounce or transliteration, but Pronounce_Translit_... must show Translit in braces when both exist.
+	// Styles with Translit_* start with Transliteration or fallback to Pronounce
+	// Styles with ...IPA... must add IPA (when exists) in braces, conditionally comma-separated in the same braces after Transliteration
```

Comment:
IPA is displayed in square brackets, not braces (same for some other lines of this pseudo-multiline comment). Also, you've updated it to not include transliteration in the same brackets.